File: src/main/java/com/cag/cagbackendapi/entities/UpcomingShowEntity.java
package com.cag.cagbackendapi.entities;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.Type;

import javax.persistence.*;
import java.util.UUID;

@Entity
@Data
@AllArgsConstructor
@NoArgsConstructor
@Table(name = "upcoming_show")
public class UpcomingShowEntity {
    @Id
    @Type(type = "pg-uuid")
    private UUID upcoming_show_id;
    private String photo_url;
    private String show_title;
    private String show_synopsis;
    private String industry_code;
    private String show_url;

    @OneToOne
    @JoinColumn(foreignKey = @ForeignKey(name = "fk_profile"))
    private ProfileEntity profileEntity;
}

File: src/main/java/com/cag/cagbackendapi/entities/UpcomingShowEntity.kt
package com.cag.cagbackendapi.entities;

import org.hibernate.annotations.GenericGenerator
import org.hibernate.annotations.Type
import java.util.*
import javax.persistence.*

@Entity
@Table(name = "upcoming_show")
data class UpcomingShowEntity (
    @Id
    @Type(type = "pg-uuid")
    @GeneratedValue(generator = "UUID")
    @GenericGenerator(
            name = "UUID",
            strategy = "org.hibernate.id.UUIDGenerator"
    )
    var upcoming_show_id: UUID?,
    var photo_url: String?,
    var show_title: String?,
    var show_synopsis: String?,
    var industry_code: String?,
    var show_url: String?,

    @OneToOne
    @JoinColumn(name = "profile_id", referencedColumnName = "profile_id")
    var profileEntity: ProfileEntity?
)
